theano/gof/cc.py

```diff
@@ -235,6 +235,7 @@ def struct_gen(args, struct_builders, blocks, sub):
 
     # declares the storage
     storage_decl = "\n".join(["PyObject* %s;" % arg for arg in args])
+    storage_init = "\n".join(["this->%s = NULL;" % arg for arg in args])
```

Comment:
Does this remove the need for the memset? I don't think so it enough to remove the memset, as it won't fix other case we cover, but it still greatly help. Op that don't have there own struct_decl (they just use params) won't need the memset.